qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -363,7 +363,86 @@ def validate_parameter(self, parameter):
 
 
 def initialize(self, params, qubits=None):
-    """Apply initialize to circuit."""
+    """Qubit initializalition is done by appending instructions to the quantum circuit (by
+    calling Initialize(params)) and the qubits we wish to initialize. Note that the
+    qubits are first set to `|0>` and then the desired state is achieved by applying
+    a state preparing unitary.
```

Comment:
```suggestion
    r"""Initialize qubits in a specific state.

    Qubit initializalition is done by first resetting the qubits to :math:`|0\rangle`
    followed by an state preparing unitary. Both these steps are included in the
    `Initialize` instruction.
```

Reply:
We usually split the docstring in a one-line summary followed by more description. I also updated the text to focus more on what's conceptually happening rather than referring to `Initialize` immediately.